### PR TITLE
Added basic documentation for iceberg support

### LINKED ISSUE #6127: Add documentation for iceberg support

### PR DESCRIPTION
**Pull request by lynnro314**
close #6127

### CODE REVIEW

**Review by rmoff — APPROVED** [review]
I've gone through and updated this doc based on my understanding - however I think it would benefit from a second pair of reviewing eyes, and since I was chatting to @arielshaqed about references in table names etc he gets tagged here ;)

File: docs/integrations/iceberg.md [hunk 1]
```diff
@@ -1,13 +1,161 @@
...
 ---
 
-# Using lakeFS with Iceberg
+# Using lakeFS with Apache Iceberg
 
-Currently, lakeFS does not support the Iceberg table format, but we have concrete 
-[plans](../roadmap.md#iceberg-support-spanhigh-priorityspan-label-label-blue-) to add it in the near future.
+{% include toc_2-3.html %}
+
+lakeFS provides its own implementation of the `SparkCatalog`, which handles the writing of the Iceberg data to lakeFS as well as reading from it and branching. Using straightforward table identifiers you can switch between branches when reading and writing data:
```
Comment (arielshaqed) [inline docs/integrations/iceberg.md:14]:
```suggestion
lakeFS provides its own implementation of `SparkCatalog`, which handles the writing of the Iceberg data to lakeFS as well as reading from it and branching. Using straightforward table identifiers you can switch between branches when reading and writing data: 
```

File: docs/integrations/iceberg.md [hunk 2]
```diff
@@ -1,13 +1,161 @@
...
+  <version>0.1.0</version>
+</dependency>
+```
+
+</div>
+<div markdown="1" id="pyspark">
+  Include the `lakefs-iceberg` jar in your package list along with Iceberg. For example: 
+
+```python
+.config("spark.jars.packages", "org.apache.iceberg:iceberg-spark-runtime-3.3_2.12:1.3.0,io.lakefs:lakefs-iceberg:0.0.1")
```
Comment (arielshaqed) [inline docs/integrations/iceberg.md:45]: This version seems inconsistent with the POM version provided above.

**Review by arielshaqed — CHANGES_REQUESTED** [review]
Neat, thanks!

**MUST** decide on which version people should run... and I'll approve!

**Review by arielshaqed — APPROVED** [review]
Yay, thanks!

**Pull request merged**